presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/HiveTransaction.java

```diff
@@ -56,12 +57,12 @@ public long getTransactionId()
     public ValidTxnWriteIdList getValidWriteIds(HiveMetastoreClosure metastore, HiveTableHandle tableHandle)
     {
         // Different calls for same table might need to lock different partitions so acquire locks every time
-        metastore.acquireSharedReadLock(
+        metastore.acquireLock(
                 identity,
                 queryId,
                 transactionId,
-                tableHandle.getPartitions().isEmpty() ? ImmutableList.of(tableHandle.getSchemaTableName()) : ImmutableList.of(),
-                tableHandle.getPartitions().orElse(ImmutableList.of()));
+                !tableHandle.getPartitions().isPresent() ? ImmutableList.of(tableHandle.getSchemaTableName()) : ImmutableList.of(),
```

Comment:
revert

Reply:
this change it needed to send correct `DataOperationType`.